FISH-788: update docs for directory config source (#126)
payara/Payara-Community-Documentation: docs/modules/ROOT/pages/documentation/microprofile/config/directory.adoc

> :source-highlighter: rouge
> 

Reviewer: ```suggestion
```
Our Antora setup doesn't support this highlighter.

Author: Done within 94b4c36

`asadmin> set-config-dir --directory=<full.path.to.dir> --target=<target[default:server]>`
*Aim*::
Sets the directory to be used for the directory config source.

> - *Leaf directory may not start with a dot*, rendering `/home/payara/.secret`

Reviewer: ```suggestion
- *Leaf directory cannot start with a dot*, rendering `/home/payara/.secret`
```

Author: Done within 94b4c36

|Mandatory

|`directory`
|Full path to the directory containing configuration files.

>  Leaf directory name may not start with a dot.

Reviewer: ```suggestion
 Leaf directory name cannot start with a dot.
```

Author: Done within 94b4c36

Docker to a directory inside a running container.

Once you configured a directory to read a config (secrets) from, you need to
make sure file names correspond to properties in your codebase.

=== Map property names to flat file hierarchy

_Since *5.183*&nbsp;_

1. Say you have two properties `property1` and `foo.bar.property2`.
2. Payara is configured with secret directory `/home/payara/secrets`
3. Secrets mounted as files to `/home/payara/secrets/property1` and
   `/home/payara/secrets/foo.bar.property2` will be read.

Remember that the files may only contain the secret or other configuration
value - no newlines, metadata, etc! Plaintext files only, no file extensions
allowed.

=== Map property names to a sub-directory structure

> _Since *5.2021.1*&nbsp;_

Author: Please change this if necessary. Can be done via GH suggestion comment for example.

1. Say you have two properties `property1` and `foo.bar.property2`.
2. Payara is configured with secret directory `/home/payara/secrets`
3. Secrets mounted as files to `/home/payara/secrets/property1` and
   `/home/payara/secrets/foo/bar/property2` will be read.

Restrictions on the files content are the same as with the flat hierarchy.

> Updates to files and subdirectories are picked up at runtime. Re-retrieving
> the config property will use the updated values. This allows for clearing

Reviewer: ```suggestion
Updates to files and subdirectories are picked up at runtime. Retrieving
the config property again will use the updated values. This allows for clearing
```

Author: Done with 94b4c36
